gregBorchers/hwWeek1.js

```diff
@@ -0,0 +1,215 @@
+
+// ### Homework #1
+// (Due Monday 1/12)
+
+// You need to turn in this homework by cloning this repo, making a new branch with your solutions, pushing to github, and issuing a pull request.
+// For details, follow the instructions [here](http://portlandcodeschool.github.io/jse/2015/01/07/command-line-and-git-slides/#/14).
+
+// Please also read the [learning objectives](objectives.md) for this week.
+
+// ---
+
+// **1)** (_Difficulty: mixed_)
+// Predict the output of each of the following expressions, and explain the rules which determine it.
+// If the output depends on the value of x, identify the conditions when the expression will output true.  Assume the cases are independent, and x is reset to an unknown value before each.
+
+// Some of these are tricky!  Don't trust your first instinct.  
+
+
+// **a)** `"1" == 1`
+true
+
+// **b)** `"1" === 1`
+false
+
+// **c)** `x == 'x'`
+true // only when x is assinged to 'x' beforehand like x = 'x'
+
+// **d)** `x == (x+'')`
+true // no matter what is assigned to x, adding '' to is is like adding nothing, so x remains the same (no side effect)
+
+// **e)** `'' == ' '`
+false // no way to make this true
+
+// **f)** `x = true`
+true // because the console returns the result of the assignment
+
+// **g)** `var x; x == 'undefined'`
+false // because x has been defined by the "var x;" statement
+
+// **h)** `'9'<'10'`
+false // the string '9' compares first position with with first position in string '10' compare so '9' is more than '1'
+
+// **i)** `typeof x + 1 === "number"`
+false // something strange is happening.  the "typeof x+1" is returning "number1" the typeof on the expression only works correctly with parens around x+1 typeof (x+1) === "number" yields true
```

Comment:
This happens because functions are higher on the precedence table than addition, so the function fires first, returning `"number"`, then the addition happens.